mods/boats/init.lua

```diff
@@ -132,35 +141,35 @@ end
 function boat.on_step(self, dtime)
 	self.v = get_v(self.object:get_velocity()) * get_sign(self.v)
 	if self.driver then
-		local driver_name = self.driver:get_player_name()
-		local ctrl = self.driver:get_player_control()
-		if ctrl.up and ctrl.down then
-			if not self.auto then
-				self.auto = true
-				minetest.chat_send_player(driver_name,
-					"[boats] Cruise on")
+		local driver_objref = minetest.get_player_by_name(self.driver)
+		if driver_objref then
```

Comment:
How about the `else` case? Didn't the previous code (similar to what is in `boat.on_detach_child` now) help to reset the driver name when it became invalid?

Reply:
Using the else case here is what caused https://github.com/minetest/minetest_game/pull/2175#issuecomment-405053172